controllers/extendeddaemonset/controller.go

```diff
@@ -510,6 +519,13 @@ func (r *Reconciler) cleanupReplicaSet(logger logr.Logger, rsList *datadoghqv1al
 	return utilserrors.NewAggregate(errs)
 }
 
+func clearFailureAnnotations(eds *datadoghqv1alpha1.ExtendedDaemonSet) {
```

Comment:
i think in this case we no longer need https://github.com/DataDog/extendeddaemonset/blob/master/pkg/plugin/canary/fail.go#L85

Reply:
Yes indeed.

Reply:
do you think it is useful to maintain a `FailedReason` annotation? it is unlikely to be seen

Reply:
I wanted to maintain "Failed" annotation to allow a user to failed manually a EDS canary

Reply:
the controller only remove annotations (paused and failed) for the auto-failed usecase.
I did it like this, because by updating the EDS.spec with the "active" ERS, it removes (delete) the canary ERS. So for me the pause and failed annotation doesn't make sense anymore. 

About the `FailedReason` we can renamed it `StatusReason`?

Reply:
apologies, i clarified my question above regarding a more general name for the function, rather than the annotation.

i understand, once the spec is updated the annotations will be stale. do you mind clarifying: 

> the controller only remove annotations (paused and failed) for the auto-failed usecase

how are we distinguishing between autofail and manual fail?

Reply:
* auto-fail is detected/trigged in the EDS-reconcile loop thanks to a CanaryFailed condition the `ERS.status.conditions`
* manual fail is detected/trigged thanks to the CanaryFailed annotation added by the user on the EDS instance.

Reply:
about your initial comments, wdyt about `clearFailureAnnotations`  -> `clearCanaryAnnotations` ?

Reply:
i see; but in either case, `IsCanaryFailed` is `true`, and this seems to be the only requirement to call `clearFailureAnnotations`

Reply:
clearFailedAnnotations are call only in the `case IsCanaryFailed`, and it correspond only to the CanaryFailed condition or annotation

for the CanaryPause condition and annotation the controller have another boolean: isCanaryPaused

Reply:
oh sorry, i misunderstood what you meant. i read 

> the controller only remove annotations (paused and failed) for the auto-failed usecase

as meaning that the manual fail case would not have the annotations removed.

that brings me back to my original question though, where I think a `FailedReason` annotation will never be observed (since it will be cleared on the first reconcile loop following the failure), unless we persist in another way (e.g. a metric tag)?